dpnp/dpnp_utils/dpnp_algo_utils.pyx

```diff
@@ -663,13 +663,19 @@ cdef tuple get_common_usm_allocation(dpnp_descriptor x1, dpnp_descriptor x2):
     return (common_sycl_queue.sycl_device, common_usm_type, common_sycl_queue)
 
 
-cdef (DPNPFuncType, void *) get_ret_type_and_func(x1_obj, DPNPFuncData kernel_data):
-    if dpnp.issubdtype(x1_obj.dtype, dpnp.integer) and not x1_obj.sycl_device.has_aspect_fp64:
+cdef (DPNPFuncType, void *) get_ret_type_and_func(device, DPNPFuncData kernel_data):
```

Comment:
Wouldn't it be better to explicitly pass only `has_aspect_fp64`, since no more information from device is required?
```suggestion
cdef (DPNPFuncType, void *) get_ret_type_and_func(DPNPFuncData kernel_data, cpp_bool has_aspect_fp64):
```

Reply:
You are right. It will be better 
thanks